src/ape_run/_cli.py

```diff
@@ -1,33 +1,152 @@
+
+            script = self.__load_module(filepath)
+            if script:
+                return script.cli  # type: ignore
+
+            return None  # NOTE: Allow other scripts to load if loading script breaks
+
+        elif "main" in code.co_names:
+            logger.debug(f"Found 'main' method in script: {relative_filepath}")
+
+            @click.command(short_help=f"Run '{relative_filepath}:main'")
+            def call():
+                module = self.__load_module(filepath)
+                if module:
+                    module.main()  # type: ignore
+
+            return call
+
+        else:
+            logger.warning(f"No 'main' method or 'cli' command in script: {relative_filepath}")
+
+            @click.command(short_help=f"Run '{relative_filepath}'")
+            def call():
+                self.__load_module(filepath)
+                # Nothing to call
+
+            return call
+
+    @cached_property
+    def _project(self) -> ProjectManager:
+        """
+        A class representing the project that is active at runtime.
+        (This is the same object as from ``from ape import project``).
 
+        Returns:
+            :class:`~ape.managers.project.ProjectManager`
+        """
 
-@click.command(cls=NetworkBoundCommand, short_help="Run scripts from the `scripts` folder")
-@click.argument("scripts", nargs=-1)
+        from ape import project
+
+        project.config.load()
+
+        return project
+
+    @cached_property
+    def commands(self) -> Dict[str, Union[click.Command, click.Group]]:
+        if not self._project.scripts_folder.exists():
+            return {}
+
+        commands = {}
+        for filepath in self._project.scripts_folder.glob("*.py"):
+            if filepath.stem.startswith("_"):
+                continue  # Ignore any "private" files
+
+            cmd = self._get_command(filepath)
+            if cmd:  # NOTE: Don't allow calling commands that failed to load
+                commands[filepath.stem] = cmd
+
+        return commands
+
+    def list_commands(self, ctx):
+        return list(sorted(self.commands))
+
+    def get_command(self, ctx, name):
+        if Path(name).exists():
+            name = Path(name).stem
+
+        if name in self.commands:
+            self.command_called = name
+            return self.commands[name]
+
+        # NOTE: don't return anything so Click displays proper error
+
+    def result_callback(self, result, interactive):
+        if interactive:
+            # TODO: Figure out how to pull out namespace for `extra_locals`
+            return console(project=self._project, extra_locals={})
+
+        return result
+
+
+@click.command(
+    cls=ScriptCommand,
+    short_help="Run scripts from the `scripts/` folder",
+)
 @click.option(
     "-i",
     "--interactive",
     is_flag=True,
     default=False,
     help="Drop into interactive console session after running",
 )
-@ape_cli_context()
```

Comment:
How does the network get set on scripts now?
I usually run ape run <script> --network <network-path>.

Am I going to have to manually add the network_option to all of my scripts that use it now?

Reply:
if we add it by default, then scripts connect to a network that don't need to

I think focusing more on "a set of functions for building scripts that integrate with ape" should be the goal here